gadget/gadget.go

```diff
@@ -353,7 +363,11 @@ func fmtIndexAndName(idx int, name string) string {
 	return fmt.Sprintf("#%v", idx)
 }
 
-func validateVolume(name string, vol *Volume) error {
+type validationState struct {
```

Comment:
I think you can combine constraints in the state to avoid passing around 2 arguments:
```
type validationState struct {
    ModelConstraints
    // set to true once system-seed role has been observed among the volume structures
    SeenSystemSeed bool
}
```

Reply:
That was my first approach in fact, but it was suggested that it should be implemented in a different way. See earlier comments above.